### PR TITLE
Allow user to delete their own contact

### PR DESCRIPTION
**Pull request by ushahidlee**
Change the visibility of the delete button, to allow a user to delete their own contact.

### CODE REVIEW

File: apps/web-mzima-client/src/app/shared/components/account-settings-modal/account-settings-modal.component.ts [hunk 1]
```diff
@@ -135,10 +135,6 @@ export class AccountSettingsModalComponent implements OnInit {
     });
   }
 
-  canDeleteContact(contact: ContactsInterface) {
-    return contact.allowed_privileges?.includes('delete');
-  }
-
```
Comment (Ifycode) [inline apps/web-mzima-client/src/app/shared/components/account-settings-modal/account-settings-modal.component.ts:141]: The remove button & modal shows up for non-admin user @ushahidlee 👍🏼
But the response from the contacts API endpoint is a 403-forbidden for such user - images below.

Question: Is there a separate fix for this already on the backend that gives such users access and that one can test with yet?

#

Screenshot 2024-02-13 at 10 51 55

#

Screenshot 2024-02-13 at 10 53 03

**Review by Ifycode — CHANGES_REQUESTED** [review]

File: apps/web-mzima-client/src/app/shared/components/account-settings-modal/account-settings-modal.component.ts (hunk 1 above)
Comment (ushahidlee) [reply apps/web-mzima-client/src/app/shared/components/account-settings-modal/account-settings-modal.component.ts:141]: Yeah there is a separate PR for backend.

https://github.com/ushahidi/platform/pull/4716

Comment (Ifycode) [reply apps/web-mzima-client/src/app/shared/components/account-settings-modal/account-settings-modal.component.ts:141]: All right then

**Review by Ifycode — APPROVED** [review]

**Review by Angamanga — APPROVED** [review]
All good 🙌

**Pull request merged**